stdlib/2and3/bz2.pyi

```diff
@@ -1,25 +1,53 @@
 import io
 import sys
-from typing import Any, IO, Optional, Union
+from os.path import _PathType
+from typing import IO, Any, Optional, TextIO, Union, overload
 
-if sys.version_info >= (3, 6):
-    from os import PathLike
-    _PathOrFile = Union[str, bytes, IO[Any], PathLike[Any]]
-elif sys.version_info >= (3, 3):
-    _PathOrFile = Union[str, bytes, IO[Any]]
+if sys.version_info >= (3, 8):
+    from typing import Literal
 else:
-    _PathOrFile = str
+    from typing_extensions import Literal
+
+_PathOrFile = Union[_PathType, IO[bytes]]
 
 def compress(data: bytes, compresslevel: int = ...) -> bytes: ...
 def decompress(data: bytes) -> bytes: ...
 
 if sys.version_info >= (3, 3):
-    def open(filename: _PathOrFile,
-             mode: str = ...,
-             compresslevel: int = ...,
-             encoding: Optional[str] = ...,
-             errors: Optional[str] = ...,
-             newline: Optional[str] = ...) -> IO[Any]: ...
+    if sys.version_info >= (3, 4):
+        # Changed in version 3.4: The 'x' (exclusive creation) mode was added.
+        _OpenBinaryMode = Literal["r", "rb", "w", "wb", "x", "xb", "a", "ab"]
+        _OpenTextMode = Literal["rt", "wt", "xt", "at"]
+    else:
+        _OpenBinaryMode = Literal["r", "rb", "w", "wb", "a", "ab"]
+        _OpenTextMode = Literal["rt", "wt", "at"]
```

Comment:
```suggestion
    _OpenBinaryMode = Literal["r", "rb", "w", "wb", "x", "xb", "a", "ab"]
    _OpenTextMode = Literal["rt", "wt", "xt", "at"]
```

Since we only support Python 2.7 and 3.5+, the conditional is unnecessary.